src/api/lib/backend/api/published.rb

```diff
@@ -7,15 +7,19 @@ class Published
       # Returns the download url for a repository
       # @return [String]
       def self.download_url_for_repository(project_name, repository_name, view = :publishedpath)
-        http_get(['/published/:project/:repository', project_name, repository_name], params: { view: view })
+        Rails.cache.fetch("download_url_for_repository-#{project_name}-#{repository_name}-#{view}") do
+          http_get(['/published/:project/:repository', project_name, repository_name], params: { view: view })
+        end
       end
 
       # Returns the build id for a repository
       # @return [String]
       def self.build_id(project_name, repository_name)
-        response = download_url_for_repository(project_name, repository_name, :status)
-        result = Xmlhash.parse(response).with_indifferent_access
-        result[:buildid]
+        Rails.cache.fetch("build_id-#{project_name}-#{repository_name}") do
+          response = download_url_for_repository(project_name, repository_name, :status)
+          result = Xmlhash.parse(response).with_indifferent_access
```

Comment:
any idea what's the point of casting the hash to a with_indifferent_access to pick one element?
`Xmlhash.parse(response)['buildid']`